staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -122,6 +122,32 @@ type JSONSchemaProps struct {
 	//      Atomic maps will be entirely replaced when updated.
 	// +optional
 	XMapType *string
+
+	// x-kubernetes-validations describes a list of validation rules for expression validation.
+	// This field is an alpha-level. Using this field requires the feature gate `CustomResourceValidationExpressions` to be enabled.
+	XValidations ValidationRules
+}
+
+// ValidationRules describe a list of validation rules for expression validation
+//  +listType=map
+//  +listMapKey=rule
+type ValidationRules []ValidationRule
+
+// ValidationRule is used for expression validation.
+type ValidationRule struct {
+	// Rule represents the validation rule which will be evaluated by CEL.
+	// ref: https://github.com/google/cel-spec
+	// The validator will be scoped to the location of the x-kubernetes-validations extension in the schema
+	// and will use `self` to represent the scoped field name.
+	// If an object property collides with a [RESERVED keyword](https://github.com/google/cel-spec/blob/v0.6.0/doc/langdef.md#syntax), it will be escaped by prepending a _ prefix.
```

Comment:
worth adding an example here.

IIUC, you can access a field named `if` as `_if` or `self._if`, right?

Reply:
Yes. I have an example added here on how to access preserved keyword. And more example in test cases. I assume we don't need to add possible examples here 😄

Reply:
I've reorganized all the tests involving escaping to make this much more clear.

Reply:
Do you mean the one for `_pre`. I think it's worth adding one more, for properties that don't initially have underscore as prefix.

Reply:
Added